lib/SymbolGraphGen/SymbolGraph.cpp

```diff
@@ -421,26 +420,12 @@ SymbolGraph::recordRequirementRelationships(Symbol S) {
   const auto VD = S.getSymbolDecl();
   if (const auto *Protocol = dyn_cast<ProtocolDecl>(VD->getDeclContext())) {
     if (VD->isProtocolRequirement()) {
+      auto Kind = RelationshipKind::RequirementOf();
+      if (VD->getAttrs().hasAttribute<OptionalAttr>())
+        Kind = RelationshipKind::OptionalRequirementOf();
       recordEdge(Symbol(this, VD, nullptr),
                  Symbol(this, Protocol, nullptr),
-                 RelationshipKind::RequirementOf());
-    }
-  }
-}
-
-void SymbolGraph::recordOptionalRequirementRelationships(Symbol S) {
```

Comment:
I don’t think tooling is prepared to susbstitute optional requirements for “regular requirements”. Right now we try to avoid attributes on graph edges, so we still need two edges for optional requirements. It’s something that might change in the future but this would probably break documentation. Please continue to call this method.

Reply:
I’ll file a separate bug to follow up on that so that optional requirements are treated the same as regular requirements.

Reply:
Do you want the separate method, or is the following fine?
```
void
SymbolGraph::recordRequirementRelationships(Symbol S) {
  const auto VD = S.getSymbolDecl();
  if (const auto *Protocol = dyn_cast<ProtocolDecl>(VD->getDeclContext())) {
    if (VD->isProtocolRequirement()) {
      auto AddEdge = [&](RelationshipKind Kind) {
        recordEdge(Symbol(this, VD, nullptr),
                   Symbol(this, Protocol, nullptr),
                   Kind);
      };
      AddEdge(RelationshipKind::RequirementOf());
      if (VD->getAttrs().hasAttribute<OptionalAttr>())
        AddEdge(RelationshipKind::OptionalRequirementOf());
    }
  }
}
```

Reply:
For readability’s sake, please keep the method.